internal/provider/provider.go

```diff
@@ -135,27 +179,34 @@ func (p *ProviderFramework) Configure(ctx context.Context, req provider.Configur
 	var data ProviderFrameworkModel
 	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
 
-	clientID := ""
-	if data.ClientID.ValueString() != "" {
-		clientID = data.ClientID.ValueString()
-	} else {
-		clientID = os.Getenv("HCP_CLIENT_ID")
+	clientConfig := clients.ClientConfig{
+		ClientID:       data.ClientID.ValueString(),
+		ClientSecret:   data.ClientSecret.ValueString(),
+		CredentialFile: data.CredentialFile.ValueString(),
+		SourceChannel:  "terraform-provider-hcp",
 	}
 
-	clientSecret := ""
-	if data.ClientSecret.ValueString() != "" {
-		clientSecret = data.ClientSecret.ValueString()
-	} else {
-		clientSecret = os.Getenv("HCP_CLIENT_SECRET")
+	// Check the environment for an HCP Service Principal
+	if clientConfig.ClientID == "" {
+		clientConfig.ClientID = os.Getenv("HCP_CLIENT_ID")
+	}
+	if clientConfig.ClientSecret == "" {
+		clientConfig.ClientSecret = os.Getenv("HCP_CLIENT_SECRET")
 	}
 
-	client, err := clients.NewClient(clients.ClientConfig{
-		ClientID:       clientID,
-		ClientSecret:   clientSecret,
-		CredentialFile: data.CredentialFile.ValueString(),
-		SourceChannel:  "terraform-provider-hcp",
-	})
+	// Read the workload_identity configuration.
+	if len(data.WorkloadIdentity.Elements()) == 1 {
+		elements := make([]WorkloadIdentityFrameworkModel, 0, 1)
+		resp.Diagnostics.Append(data.WorkloadIdentity.ElementsAs(ctx, &elements, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		clientConfig.WorloadIdentityTokenFile = elements[0].TokenFile.ValueString()
+		clientConfig.WorkloadIdentityResourceName = elements[0].ResourceName.ValueString()
+	}
```

Comment:
Should there be an error if more than one `WorkloadIdentity` element is provided?